packages/types/index.d.ts

```diff
@@ -1,12 +1,12 @@
 import { Action } from 'redux'
 
-export type Saga<Args extends any[] = any[]> = (...args: Args) => IterableIterator<any>
+export type Saga<Args extends any[] = any[]> = (...args: Args) => Iterator<any>
 
 /**
  * Annotate return type of generators with `SagaIterator` to get strict
  * type-checking of yielded effects.
  */
-export type SagaIterator = IterableIterator<StrictEffect>
+export type SagaIterator<RT = any> = Iterator<StrictEffect, RT>
```

Comment:
I found I needed this to be `Iterator<StrictEffect, RT, any>`.

Reply:
Try the following with and without that change:
```
type RT<A extends Effect<any, any>> = 
  A extends Effect<any, CallEffectDescriptor<infer RT>> ? RT :
  A extends Effect<any, CombinatorEffectDescriptor<() => SagaIterator<infer RT>>> ? RT :
  unknown;

function* aSaga(): SagaIterator<number> {
  const effect = call(() => Promise.resolve(1));
  const effectResult: RT<typeof effect> = yield effect;
  
  return effectResult;
}
```

Reply:
Wow, what neat idea to strongly type effects return type you gave here 🧐 . Thanks.

Yeah, TNext is of type unknown by default. Any is probably our best option.

Reply:
Thanks. I also found I had to go with `Iterator<Effect, RT, any>` (not `Strict`) to get this to work:

```
export function* raceArrayExample(): SagaIterator<number> {
  const effect = race([shouldInferNumber, shouldInferNumber]);
  const effectResult: RT<typeof effect> = yield effect;

  return Math.max(44, effectResult);
}

export function* raceObjectExample(): SagaIterator<number> {
  const effect = race({one: shouldInferNumber, two: shouldInferNumber});
  const effectResult: RT<typeof effect> = yield effect;

  return Math.max(45, effectResult);
}
```